### PR TITLE
CUDNN docstring revisions.

### PR DESCRIPTION
**Pull request by GunnarFarneback**
Docstring changes only. This mostly adds a lot of backticks but there are some other revisions as well. @denizyuret, if there's something you don't agree with, point it out and I'll revert it.

### CODE REVIEW

File: lib/cudnn/multiheadattn.jl [hunk 1]
```diff
@@ -40,8 +40,8 @@ Keyword arguments describing the attention operation when `d` is not given:
...
-* `currIdx::Integer = -1`: Time-step (0-based) in queries to process. When the currIdx argument is negative, all Q time-steps are processed. When currIdx is zero or positive, the forward response is computed for the selected time-step only. The latter input can be used in inference mode only, to process one time-step while updating the next attention window and Q, R, K, V inputs in-between calls.
-* `loWinIdx, hiWinIdx::Array{Cint}`: Two host integer arrays specifying the start and end (0-based) indices of the attention window for each Q time-step. The start index in K, V sets is inclusive, and the end index is exclusive. By default set at 0 and `kvMaxSeqLength` respectively.
+* `currIdx::Integer = -1`: Time-step (0-based) in queries to process. When the `currIdx` argument is negative, all ``Q`` time-steps are processed. When `currIdx` is zero or positive, the forward response is computed for the selected time-step only. The latter input can be used in inference mode only, to process one time-step while updating the next attention window and ``Q``, ``R``, ``K``, ``V`` inputs in-between calls.
```
Comment (GunnarFarneback) [inline lib/cudnn/multiheadattn.jl]: There is no previous mention of ``R``. I don't know this operation so I can't say whether ``R`` should just be removed here or documented elsewhere.

File: lib/cudnn/rnn.jl [hunk 2]
```diff
@@ -10,28 +10,28 @@ to input `x`.
...
+* `inputSize::Integer = size(x,1)`: Size of the input vector in the RNN model. When the `inputMode=CUDNN_SKIP_INPUT`, the `inputSize` should match the `hiddenSize` value.
+* `projSize::Integer = hiddenSize`: The size of the LSTM cell output after the recurrent projection. This value should not be larger than `hiddenSize`. It is legal to set `projSize` equal to `hiddenSize`, however, in this case, the recurrent projection feature is disabled. The recurrent projection is an additional matrix multiplication in the LSTM cell to project hidden state vectors `ht` into smaller vectors `rt = Wr * ht`, where `Wr` is a rectangular matrix with `projSize` rows and hiddenSize columns. When the recurrent projection is enabled, the output of the LSTM cell (both to the next layer and unrolled in-time) is `rt` instead of `ht`. The recurrent projection can be enabled for LSTM cells and `CUDNN_RNN_ALGO_STANDARD` only.
```
Comment (GunnarFarneback) [inline lib/cudnn/rnn.jl:26]: It looked like `Wrht` was supposed to be `Wr * ht` but it's really just a guess.

File: lib/cudnn/multiheadattn.jl (hunk 1 above)
Comment (denizyuret) [reply lib/cudnn/multiheadattn.jl]: Here is the original source: https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnMultiHeadAttnForward

After reading it again, I think you are correct: ``R`` is a typo and can be removed.

File: lib/cudnn/rnn.jl (hunk 2 above)
Comment (denizyuret) [reply lib/cudnn/rnn.jl:26]: Here is the original source: https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnRNNForward

This was a hasty copy-paste issue, your guess is correct, it should be `Wr * ht`, thanks for catching this.

**Review by denizyuret — APPROVED** [review]
Thanks for going over the docs @GunnarFarneback. The two issues you commented about were copy/paste issues from the original cudnn docs. I gave pointers to the original source and my best guesses.

File: lib/cudnn/multiheadattn.jl (hunk 1 above)
Comment (GunnarFarneback) [reply lib/cudnn/multiheadattn.jl]: I agree that ``R`` is definitely a typo but it seems plausible that it could be a typo for ``O``.

Comment (denizyuret) [reply lib/cudnn/multiheadattn.jl]: I suspected the same thing. However in the more detailed description above (https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnMultiHeadAttnForward paragraph starting with "In the inference mode") it only mentions updating `Q,K,V` vectors. Also `O` is for output, i.e. the caller does not initialize it, only the return value is written over it.

Comment (GunnarFarneback) [reply lib/cudnn/multiheadattn.jl]: Ok, let's just remove it.

**Comment by maleadt** [comment]
Thanks!

**Pull request merged**